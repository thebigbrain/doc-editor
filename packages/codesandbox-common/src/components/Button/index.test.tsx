import 'jest-styled-components';
import React from 'react';
import mountWithTheme from '../../test/themeMount';
import { Button } from '.';

describe('<Button /> rendering', () => {
  it('no props', () => {
    expect(mountWithTheme(<Button>Le button</Button>)).toMatchSnapshot();
  });
  it('Danger Button', () => {
    expect(mountWithTheme(<Button danger>Le button</Button>)).toMatchSnapshot();
  });
  it('Secondary Button', () => {
    expect(
      mountWithTheme(<Button secondary>Le button</Button>)
    ).toMatchSnapshot();
  });
  it('Small Button', () => {
    expect(mountWithTheme(<Button small>Le button</Button>)).toMatchSnapshot();
  });
  it('block Button', () => {
    expect(mountWithTheme(<Button block>Le button</Button>)).toMatchSnapshot();
  });

  it('Href Button', () => {
    expect(
      mountWithTheme(<Button href="#">Le button</Button>)
    ).toMatchSnapshot();
  });
});
