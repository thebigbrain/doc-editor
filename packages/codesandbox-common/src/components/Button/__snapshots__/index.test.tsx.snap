// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> rendering Danger Button 1`] = `ReactWrapper {}`;

exports[`<Button /> rendering Href Button 1`] = `ReactWrapper {}`;

exports[`<Button /> rendering Secondary Button 1`] = `ReactWrapper {}`;

exports[`<Button /> rendering Small Button 1`] = `ReactWrapper {}`;

exports[`<Button /> rendering block Button 1`] = `ReactWrapper {}`;

exports[`<Button /> rendering no props 1`] = `ReactWrapper {}`;
